service/controller/azure/v22/resource_set.go

```diff
@@ -502,3 +506,22 @@ func toCRUDResource(logger micrologger.Logger, ops controller.CRUDResourceOps) (
 
 	return r, nil
 }
+
+func getSubscriptionId(obj interface{}, k8s kubernetes.Interface) (string, error) {
+	cr, err := key.ToCustomObject(obj)
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	secret, err := k8s.CoreV1().Secrets(cr.Spec.Guest.CredentialSecret.Namespace).Get(cr.Spec.Guest.CredentialSecret.Name, metav1.GetOptions{})
```

Comment:
We always use the `key` package for accessing CRs. So you should add `key.CredentialSecretName`.